src/main/java/org/wso2/carbon/inbound/iso8583/listening/ISO8583MessageConnection.java

```diff
@@ -0,0 +1,168 @@
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.Properties;
+import java.util.concurrent.*;
+
+/**
+ * Class for maintain the socket connections.
+ */
+public class ISO8583MessageConnection extends Thread {
+    private static final Log log = LogFactory.getLog(ISO8583MessageConnection.class);
+    private int port;
+    private ServerSocket server;
+    private InboundProcessorParams params;
+    private ExecutorService threadPool;
+    private boolean listening = false;
+
+    public ISO8583MessageConnection(int port, InboundProcessorParams params) {
+        this.port = port;
+        this.params = params;
+        getExecutorService();
+    }
+
+    /**
+     * create the threadPool to handle the concurrent request.
+     */
+    public void getExecutorService() {
+        Properties properties = params.getProperties();
+        String coreThreads = properties.getProperty(ISO8583Constant.INBOUND_CORE_THREADS);
+        String maxThreads = properties.getProperty(ISO8583Constant.INBOUND_MAX_THREADS);
+        String threadSafeTime = properties.getProperty(ISO8583Constant.INBOUND_THREAD_ALIVE);
+        String queueLength = properties.getProperty(ISO8583Constant.INBOUND_THREAD_QLEN);
+        try {
+            if ((!StringUtils.isEmpty(coreThreads)) && (!StringUtils.isEmpty(maxThreads)) &&
+                    (!StringUtils.isEmpty(threadSafeTime)) && (!StringUtils.isEmpty(queueLength))) {
+                BlockingQueue<Runnable> workQueue = new LinkedBlockingQueue<Runnable>(Integer.parseInt(queueLength));
+                threadPool = new ThreadPoolExecutor(Integer.parseInt(coreThreads), Integer.parseInt(maxThreads)
+                        , Integer.parseInt(threadSafeTime), TimeUnit.SECONDS, workQueue);
+            } else {
+                BlockingQueue<Runnable> workQueue = new LinkedBlockingQueue<Runnable>(
+                        Integer.parseInt(ISO8583Constant.THREAD_QLEN));
+                threadPool = new ThreadPoolExecutor(Integer.parseInt(ISO8583Constant.CORE_THREADS),
+                        Integer.parseInt(ISO8583Constant.MAX_THREADS), Integer.parseInt(ISO8583Constant.KEEP_ALIVE),
+                        TimeUnit.SECONDS, workQueue);
+            }
+        } catch (NumberFormatException e) {
+            handleException("One of the property or properties of thread specified is of an invalid type", e);
+        }
+    }
+
+    /**
+     * create the server socket which is to accept a connection from a client.
+     */
+    private void createConnection() throws IOException {
+        server = new ServerSocket(port);
+        log.info("Server is listening on port :" + port);
+        while (!listening) {
+            Socket socketConnection = server.accept();
+            log.debug("Client connected to socket: " + socketConnection.toString());
+            handleClientRequest(socketConnection, params);
+        }
+    }
+
+    public void run() {
+        try {
+            createConnection();
```

Comment:
Need to improve this. only need to retry if something goes wrong while accepting connections.